From: Jay Bratcher <jayb@cox-internet.com>
Newsgroups: comp.os.linux.hardware
Subject: Re: ATI Radeon 9000
Date: Sun, 15 Sep 2002 13:07:51 -0400
Organization: Posted via Supernews, http://www.supernews.com
Message-ID: <3D84BE67.40300@cox-internet.com>
User-Agent: Mozilla/5.0 (X11; U; Linux i686; en-US; rv:0.9.8) Gecko/20020204
X-Accept-Language: en-us
MIME-Version: 1.0
References: <3d844ae3$0$177$9b622d9e@news.freenet.de>
Content-Type: multipart/mixed;
 boundary="------------060008020003010106020906"
X-Complaints-To: abuse@supernews.com
Lines: 162


diff -ruN xc.orig/programs/Xserver/hw/xfree86/common/xf86PciInfo.h xc/programs/Xserver/hw/xfree86/common/xf86PciInfo.h
--- xc.orig/programs/Xserver/hw/xfree86/common/xf86PciInfo.h	Tue Jan 15 21:00:43 2002
+++ xc/programs/Xserver/hw/xfree86/common/xf86PciInfo.h	Sat Sep  7 17:03:15 2002
@@ -253,6 +253,7 @@
 #define PCI_CHIP_R200_QN	0x514E
 #define PCI_CHIP_R200_QO	0x514F
 #define PCI_CHIP_RV200_QW	0x5157
+#define PCI_CHIP_RV250_If	0x4966
 #define PCI_CHIP_RADEON_QY	0x5159
 #define PCI_CHIP_RADEON_QZ	0x515A
 #define PCI_CHIP_R200_Ql	0x516C
@@ -890,6 +891,7 @@
 				{PCI_CHIP_R200_QN,	"Radeon 8500 QN",0},
 				{PCI_CHIP_R200_QO,	"Radeon 8500 QO",0},
 				{PCI_CHIP_RV200_QW,	"Radeon 7500 QW",0},
+				{PCI_CHIP_RV250_If,	"Radeon 9000 If",0},
 				{PCI_CHIP_RADEON_QY,	"Radeon VE QY",0},
 				{PCI_CHIP_RADEON_QZ,	"Radeon VE QZ",0},
 				{PCI_CHIP_R200_Ql,	"Radeon 8500 Ql",0},
diff -ruN xc.orig/programs/Xserver/hw/xfree86/drivers/ati/atichip.c xc/programs/Xserver/hw/xfree86/drivers/ati/atichip.c
--- xc.orig/programs/Xserver/hw/xfree86/drivers/ati/atichip.c	Wed Jan 16 11:22:25 2002
+++ xc/programs/Xserver/hw/xfree86/drivers/ati/atichip.c	Fri Sep  6 15:49:54 2002
@@ -617,6 +617,9 @@
         case NewChipID('Q', 'W'):
             return ATI_CHIP_RV200;
 
+	case NewChipID('I', 'f'):
+	    return ATI_CHIP_RV250;
+
         case NewChipID('H', 'D'):
             return ATI_CHIP_HDTV;
 
diff -ruN xc.orig/programs/Xserver/hw/xfree86/drivers/ati/atichip.h xc/programs/Xserver/hw/xfree86/drivers/ati/atichip.h
--- xc.orig/programs/Xserver/hw/xfree86/drivers/ati/atichip.h	Wed Jan 16 11:22:25 2002
+++ xc/programs/Xserver/hw/xfree86/drivers/ati/atichip.h	Fri Sep  6 15:50:24 2002
@@ -92,6 +92,7 @@
     ATI_CHIP_RADEONMOBILITY7,   /* Radeon M7 */
     ATI_CHIP_R200,              /* R200 */
     ATI_CHIP_RV200,             /* RV200 */
+    ATI_CHIP_RV250,             /* RV250 */
     ATI_CHIP_HDTV               /* HDTV */
 } ATIChipType;
 
diff -ruN xc.orig/programs/Xserver/hw/xfree86/drivers/ati/atiprobe.c xc/programs/Xserver/hw/xfree86/drivers/ati/atiprobe.c
--- xc.orig/programs/Xserver/hw/xfree86/drivers/ati/atiprobe.c	Fri Jan 18 11:56:16 2002
+++ xc/programs/Xserver/hw/xfree86/drivers/ati/atiprobe.c	Fri Sep  6 15:51:05 2002
@@ -1734,6 +1734,7 @@
                     case ATI_CHIP_RADEONMOBILITY7:
                     case ATI_CHIP_R200:
                     case ATI_CHIP_RV200:
+		    case ATI_CHIP_RV250:
                         DoRadeon = TRUE;
                         continue;
 
diff -ruN xc.orig/programs/Xserver/hw/xfree86/drivers/ati/radeon.h xc/programs/Xserver/hw/xfree86/drivers/ati/radeon.h
--- xc.orig/programs/Xserver/hw/xfree86/drivers/ati/radeon.h	Wed Nov 14 11:50:44 2001
+++ xc/programs/Xserver/hw/xfree86/drivers/ati/radeon.h	Fri Sep  6 15:52:00 2002
@@ -268,6 +268,7 @@
     BOOL              HasCRTC2;     /* VE/M6/M7 */
     BOOL              IsR200;       /* R200 chip */
     BOOL              IsRV200;      /* RV200 chip */
+    BOOL              IsRV250;      /* RV250 chip */
     BOOL              IsSecondary;  /* second Screen */
     BOOL              UseCRT;       /* force use CRT port as primary */
     BOOL              IsM6;         /* M6 card, for some workarounds */
diff -ruN xc.orig/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c
--- xc.orig/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c	Tue Jan 15 21:00:43 2002
+++ xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c	Fri Sep  6 15:54:28 2002
@@ -1146,6 +1146,11 @@
             info->HasCRTC2 = TRUE;  
             info->IsRV200 = TRUE;
             break;
+	case PCI_CHIP_RV250_If:  /* RV250 */
+	     info->HasCRTC2 = TRUE;  
+	     info->IsRV250 = TRUE;
+	     break;
+
         default: 
             info->HasCRTC2 = FALSE;  
     }
@@ -1285,6 +1290,7 @@
 	case PCI_CHIP_R200_Ql:
 	case PCI_CHIP_R200_BB:
 	case PCI_CHIP_RV200_QW:
+	case PCI_CHIP_RV250_If:
 	default:                 info->IsPCI = FALSE; break;
 	}
     }
diff -ruN xc.orig/programs/Xserver/hw/xfree86/drivers/ati/radeon_probe.c xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_probe.c
--- xc.orig/programs/Xserver/hw/xfree86/drivers/ati/radeon_probe.c	Tue Jan 15 21:00:44 2002
+++ xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_probe.c	Fri Sep  6 15:56:29 2002
@@ -89,6 +89,7 @@
     { PCI_CHIP_R200_Ql, "ATI Radeon 8500 Ql (AGP)" },
     { PCI_CHIP_R200_BB, "ATI Radeon 8500 BB (AGP)" },
     { PCI_CHIP_RV200_QW, "ATI Radeon 7500 QW (AGP)" },
+    { PCI_CHIP_RV250_If, "ATI Radeon 9000 If (AGP)" },
     { -1,                 NULL }
 };
 
@@ -108,6 +109,7 @@
     { PCI_CHIP_R200_Ql, PCI_CHIP_R200_Ql, RES_SHARED_VGA },
     { PCI_CHIP_R200_BB, PCI_CHIP_R200_BB, RES_SHARED_VGA },
     { PCI_CHIP_RV200_QW, PCI_CHIP_RV200_QW, RES_SHARED_VGA },
+    { PCI_CHIP_RV250_If, PCI_CHIP_RV250_If, RES_SHARED_VGA },
     { -1,                 -1,                 RES_UNDEFINED }
 };
 
@@ -240,6 +242,7 @@
            pEnt->chipset == PCI_CHIP_R200_Ql ||
            pEnt->chipset == PCI_CHIP_R200_BB ||
            pEnt->chipset == PCI_CHIP_RV200_QW ||
+	   pEnt->chipset == PCI_CHIP_RV250_If ||
            pEnt->chipset == PCI_CHIP_RADEON_LW ||
            pEnt->chipset == PCI_CHIP_RADEON_LY ||
            pEnt->chipset == PCI_CHIP_RADEON_LZ)

--

